cocoa/tests_backend/widgets/base.py

```diff
@@ -0,0 +1,44 @@
+# from System import EventArgs, Object
+
+from .properties import toga_color
+
+
+class SimpleProbe:
+    def __init__(self, widget):
+        self.native = widget._impl.native
+        assert isinstance(self.native, self.native_class)
+
+    def assert_container(self, container):
+        container_native = container._impl.native
+        for control in container_native.subviews:
+            if control == self.native:
+                break
+        else:
+            raise ValueError(f"cannot find {self.native} in {container_native}")
+
+    @property
+    def enabled(self):
+        return self.native.enabled
+
+    @property
+    def background_color(self):
+        return toga_color(self.native.backgroundColor)
+
+    @property
+    def color(self):
+        return toga_color(self.native.textColor)
```

Comment:
On the other platforms I tried to make the probe hierarchy follow the native widget hierachy. For example, on Windows the base Control class has a ForeColor property, so the SimpleProbe implements the `color` method. But on Android there is no such property, so `SimpleProbe.color` calls `skip`, and the widget-specific probes override the method where applicable.

If I understand the Cocoa documentation correctly, there are no color properties in the base NSView class, so we should take the second approach.

Reply:
But maybe I've misunderstood something, because if NSButton doesn't have a `backgroundColor` property, then how did it pass `test_button.py::test_background_color`?

Reply:
I guess that's what [this page](https://developer.apple.com/documentation/appkit/nsbutton) calls a "cover" method for NSButtonCell. But I still can't see where `drawsBackground` is coming from.

Reply:
Organising probe APIs to match the underlying platforms makes sense. It does open up an interesting gap where some platforms will have a "color" API on the base widget, but on others it will be platform specific; but I guess that will manifest as a test failure/error due to missing a required probe API.

You're correct that `backgroundColor` is a cover API; however, it looks like the base widget usage of `drawsBackground` is an accidental pass. `drawsBackground` only exists on a few widgets (NSTextField and NSScrollView being the most notable); it doesn't exist on NSButton (even as a cover method), but you can *set* an arbitrary property on an Rubicon ObjC object and it will just be annotated on the object.

However - this becomes an interesting case of somewhere that checking properties won't necessarily tell us that something is working correctly. This is the current output of the `colors` example on macOS, with a background color of green, and a color of blue.

<img width="812" alt="Screen Shot 2022-12-09 at 9 42 58 am" src="https://user-images.githubusercontent.com/37345/206603950-29a42bf0-20f9-49ed-8aa7-3b0935ab3d64.png">

The button does have something that could be called the background color altered, and the `backgroundColor` property is correct; but it's not changing the background color (at least, not in the way you'd expect). 

I guess this comes down to verifying that the behaviour (and the probe) is actually correct at the time of implementation.

Reply:
Yes, it'll be the responsibility of the probes to check whatever native properties are appropriate for each widget.

Reply:
For future reference: see [related discussion below](https://github.com/beeware/toga/pull/1707#discussion_r1071602317).